application/templates/cronjob.yaml

```diff
@@ -39,11 +39,17 @@ spec:
           {{- end }}
           containers:
           - name: {{ $name }}
+            {{- if $job.image.tag }}
             {{- if $job.image.digest }}
-            image: "{{ $job.image.repository }}@{{ $job.image.digest }}"
-            {{- else if $job.image.tag }}
+            image: "{{ $job.image.repository }}:{{ $job.image.tag }}@{{ $job.image.digest }}"
+            {{- else }}
             image: "{{ $job.image.repository }}:{{ $job.image.tag }}"
             {{- end }}
+            {{- else if $job.image.digest }}
+            image: "{{ $job.image.repository }}@{{ $job.image.digest }}"
+            {{- else }}
+            image: "{{ $job.image.repository }}"
+            {{- end }}
```

Comment:
To avoid repetition, here's a suggestion:

```gotpl
{{- $image := $job.image.repository }}
{{- with $job.image.tag }}{{ $image = print $image ":" . }}{{ end }}
{{- with $job.image.digest }}{{ $image = print $image "@" . }}{{ end }}
image: {{ $image }}
```

I do believe, however, that if an image repository isn't specified, we should have a templating error, for containers can't be without an image:

```gotpl
{{- $image := $job.image.repository }}
{{- if not $image }}{{ print "Undefined image at CronJob definition at: " $name | fail }}{{ end }}
{{- with $job.image.tag }}{{ $image = print $image ":" . }}{{ end }}
{{- with $job.image.digest }}{{ $image = print $image "@" . }}{{ end }}
image: {{ $image }}
```

If we go this route, there should be a unit test for it!

Reply:
It would be possible to remove the default for `image.repository` via

`{{ required "A valid value for .Values.repository.image is required!" .Values.image.repository }}`

but since the default was already set to `repository/image-name` i did not change this behaviour.

Reply:
Looks like generic default values like those could be refactored out to allow for these safeguards to be put in place. That's probably outside the scope of this pull request, however.